otelcol/collector.go

```diff
@@ -140,7 +143,13 @@ func NewCollector(set CollectorSettings) (*Collector, error) {
 		configProvider:             configProvider,
 		bc:                         bc,
 		updateConfigProviderLogger: cc.SetCore,
-	}, nil
+	}
+	reloadServer, err := col.newReloadServer()
```

Comment:
Assigning a variable generated by the struct into the struct is a bit weird.
Why isn't the struct directly setting the value on itself?